### ISSUE TITLE
HarmonyHub not initialising

### ISSUE THREAD

**Issue by lloydwood** [KraigM/homebridge-harmonyhub#50]
Not sure what's happening here. 

I get [Harmony] Initializing HarmonyHub platform . . .

And then nothing. I'm assuming this means it didn't initialise because it doesn't seem to get on to actually searching/connecting with the Harmony Hub. 

I think I have my config.json set up right:

{
    "bridge": {
        "name": "Homebridge",
        "username": "CC:22:3D:E3:CE:30",
        "port": 51826,
        "pin": "031-45-154"
    },
    
    "description": "This is an example configuration file with one fake accessory and one fake pla
tform. You can use this as a template for creating your own configuration file containing devices 
you actually own.",
    
    "accessories": [],

    "platforms": [
        {
            "platform" : "HarmonyHub",
            "name" : "Harmony",
            "ip_address" : "192.168.0.146"
        }
    ]
}

**Comment by RXM307** [comment]
I'm seeing the same thing, running tcpdump I cant even see it trying to connect to the ip_address

**Comment by taconugget** [comment]
I too am having the same issue.

**Comment by taconugget** [comment]
After further playing I have disabled my UFW firewall for testing purposes and it appears to be the issue. With ufw disabled I now see this. There must be a port we need to open in addition to 51826.

```
[9/29/2016, 2:37:40 PM] Homebridge is running on port 51826.
[9/29/2016, 2:37:40 PM] [Living Room Harmony Hub] Found activities:
	Watch TV
	OSMC
	Listen to Music
	Watch TV Stereo
	Stream Music
	Watch Apple TV
	Watch a Movie
	PowerOff
	Play PS4
	Wii U
```

**Comment by taconugget** [comment]
So... after finding many logs in my UFW.log file like this:

```
Sep 29 14:30:34 odroid64 kernel: [68144.920730] [UFW BLOCK] IN=eth0 OUT= MAC=00:1e:06:33:5a:37:00:04:20:e8:d3:6a:08:00 SRC=192.168.1.9 DST=192.168.1.205 LEN=60 TOS=0x00 PREC=0x00 TTL=64 ID=23443 DF PROTO=TCP SPT=38294 DPT=61991 WINDOW=5840 RES=0x00 SYN URGP=0
Sep 29 14:30:53 odroid64 kernel: [68163.949119] [UFW BLOCK] IN=eth0 OUT= MAC=00:1e:06:33:5a:37:00:04:20:e8:d3:6a:08:00 SRC=192.168.1.9 DST=192.168.1.205 LEN=60 TOS=0x00 PREC=0x00 TTL=64 ID=45502 DF PROTO=TCP SPT=38300 DPT=61991 WINDOW=5840 RES=0x00 SYN URGP=0
Sep 29 14:31:12 odroid64 kernel: [68182.985177] [UFW BLOCK] IN=eth0 OUT= MAC=00:1e:06:33:5a:37:00:04:20:e8:d3:6a:08:00 SRC=192.168.1.9 DST=192.168.1.205 LEN=60 TOS=0x00 PREC=0x00 TTL=64 ID=31548 DF PROTO=TCP SPT=38306 DPT=61991 WINDOW=5840 RES=0x00 SYN URGP=0
Sep 29 14:31:34 odroid64 kernel: [68205.212718] [UFW BLOCK] IN=eth0 OUT= MAC=00:1e:06:33:5a:37:00:04:20:e8:d3:6a:08:00 SRC=192.168.1.9 DST=192.168.1.205 LEN=60 TOS=0x00 PREC=0x00 TTL=64 ID=58381 DF PROTO=TCP SPT=38313 DPT=61991 WINDOW=5840 RES=0x00 SYN URGP=0
```
and @RXM307 tcpdump file:

`Tue, 27 Sep 2016 01:01:53 GMT harmonyhubjs:discover:responsecollector ResponseCollector(61991)`

I have enabled UFW and opened up the port 61991. Doing this seems to fix the issue.

`sudo ufw enable`
`sudo ufw allow 61991`

```
[9/29/2016, 2:42:21 PM] [Living Room Harmony Hub] Fetching Logitech Harmony activities...
[9/29/2016, 2:42:23 PM] [Living Room Harmony Hub] Found activities:
	Watch TV
	OSMC
	Listen to Music
	Watch TV Stereo
	Stream Music
	Watch Apple TV
	Watch a Movie
	PowerOff
	Play PS4
	Wii U

```

**Comment by RXM307** [comment]
I just checked and there is no firewall enabled

pi@pi ~ $ sudo iptables -L
Chain INPUT (policy ACCEPT)
target     prot opt source               destination

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination


running in debug doesn't really show anymore detail and no activities returned :(

[9/30/2016, 9:45:33 AM] Loaded plugin: homebridge-harmonyhub
[9/30/2016, 9:45:42 AM] Registering platform 'homebridge-harmonyhub.HarmonyHub'
[9/30/2016, 9:45:42 AM] ---
[9/30/2016, 9:45:42 AM] Loaded config.json with 0 accessories and 1 platforms.
[9/30/2016, 9:45:42 AM] ---
[9/30/2016, 9:45:43 AM] Loading 1 platforms...
[9/30/2016, 9:45:43 AM] [Lounge Room Harmony Hub] Initializing HarmonyHub platform...
  harmonyhubjs:discover:explorer Explorer(61991) +0ms
  harmonyhubjs:discover:ping Ping(61991) +25ms
  harmonyhubjs:discover:explorer start() +57ms
  harmonyhubjs:discover:responsecollector ResponseCollector(61991) +6ms
  harmonyhubjs:discover:responsecollector start() +22ms
  harmonyhubjs:discover:ping start() +115ms
[9/30/2016, 9:45:43 AM] Loading 0 accessories...
Scan this code with your HomeKit App on your iOS device to pair with Homebridge:

    ┌────────────┐
    │ ... │
    └────────────┘

[9/30/2016, 9:45:43 AM] Homebridge is running on port 51826.
[9/30/2016, 9:45:43 AM] [Lounge Room Harmony Hub] Finalizing Plugin Launch
  EventedHTTPServer Server listening on port 51826 +0ms
  harmonyhubjs:discover:ping emit() +1s
  harmonyhubjs:discover:ping emit() +1s
  harmonyhubjs:discover:ping emit() +1s
  harmonyhubjs:discover:ping emit() +1s
  harmonyhubjs:discover:explorer executeCleanUp() +843ms
  harmonyhubjs:discover:ping emit() +158ms

**Comment by taconugget** [comment]
Have you tried calling out the IP address of you harmony hub in the config.json?

I'd also look at the wifi access point / router and make sure it can pass the packets. 

I'm a newbie but it looks like it's sending a request for any harmony to answer and it's not receiving one back.

**Comment by RXM307** [comment]
Yes I've already got the ip_address in my config.js (I've also tried it without the ip_address, it also does not work), if I nmap the IP address of the harmony Hub I get the following.

Host is up (0.018s latency).
Not shown: 997 closed ports
PORT     STATE SERVICE
5222/tcp open  xmpp-client
8088/tcp open  radan-http
8222/tcp open  unknown

is there anyway I can poll the hub directly? from my limited knowledge it looks like the homebridge-harmonyhub plugin communicates on port 5222.

**Comment by RXM307** [comment]
OK found my issue, my WiFi did not have broadcast forwarding enabled, perhaps @lloydwood has the same issue?

**Comment by RXM307** [comment]
@KraigM  just saw your release notes on the 0.2.0 release, no broadcast forwarding is one reason why people may want to specify the IP rather then letting discovery fail, although when it wont connect anyway I can see why you disabled it.

**Comment by djrobx** [comment]
Having same issue.   0.1.1 works perfectly fine with a specified IP, but auto-detect and thus 2.0 is broken.   I have a DD-WRT wireless bridge but no meddling with multicast options seems to help.

Please add manually specified IPs back again.
